### PR TITLE
Add support for the IntlMessageFormatter

### PR DESCRIPTION
**Pull request by digilist**
This adds support for the new ICU Message Format that was introduced with Symfony 4.2. I tested the plugin and I was getting auto completion, could insert new keys directly from template and got suggestions for the available placeholders.

The parsing of the available placeholders is rather simple and looks only for `{var}` and not for more complex ICU formats.

One issue I am aware of, is that Symfony registers registers the contents of a `message+intl-icu` file inside the `message` and `message+intl-icu` domains. So there are actually two domains, even though the intl-icu translations are append to the main domain. However, I do not see any reason why someone would access the intl-icu translation directly and loose the formatting, so I didn't consider this case.

Further information:
https://symfony.com/blog/new-in-symfony-4-2-intlmessageformatter
symfony/symfony#28952
#1251

### CODE REVIEW

File: src/main/java/fr/adrienbrault/idea/symfony2plugin/translation/parser/TranslationStringParser.java [hunk 1]
```diff
@@ -51,10 +51,10 @@ public TranslationStringMap parse(File file) {
 
         TranslationStringMap string_map = new TranslationStringMap();
 
-        Matcher matcher = Pattern.compile("'([\\w]+)'\\s=>.*?array\\s\\((.*?)\\s\\),", Pattern.MULTILINE).matcher(string);
+        Matcher matcher = Pattern.compile("'([\\w]+(\\+intl\\-icu)?)'\\s=>.*?array\\s\\((.*?)\\s\\),", Pattern.MULTILINE).matcher(string);
```
Comment (digilist) [inline src/main/java/fr/adrienbrault/idea/symfony2plugin/translation/parser/TranslationStringParser.java]: I just noticed, that the TranslationStringParser isn't used anywhere, do I see this correctly? In that case my changes here do not have any effect and the classes can be removed?

File: src/main/java/fr/adrienbrault/idea/symfony2plugin/stubs/indexes/TranslationStubIndex.java [hunk 2]
```diff
@@ -139,7 +139,13 @@ private String getDomainName(@NotNull String fileName) {
                     return null;
                 }
 
-                return StringUtils.join(Arrays.copyOfRange(split, 0, split.length - 2), ".");
+                String domain = StringUtils.join(Arrays.copyOfRange(split, 0, split.length - 2), ".");
+
+                // Remove +intl-icu suffix, as it is not part of the domain
+                // https://symfony.com/blog/new-in-symfony-4-2-intlmessageformatter
+                domain = domain.replace("+intl-icu", "");
```
[the remark this answers is not in the document]
Comment (Haehnchen) [reply src/main/java/fr/adrienbrault/idea/symfony2plugin/stubs/indexes/TranslationStubIndex.java]: maybe check for endWiths "+intl-icu" and then remove the ending chars, makes it more tolerate to naming

File: src/main/java/fr/adrienbrault/idea/symfony2plugin/translation/dict/TranslationUtil.java [hunk 3]
```diff
@@ -402,6 +402,15 @@ public static boolean isSupportedXlfFile(@NotNull PsiFile psiFile) {
             placeholder.add(matcher.group(1));
         }
 
+        // Simple parser for MessageFormat by the ICU project
+        matcher = Pattern.compile("(\\{\\s*[^{]*\\s*})").matcher(text);
+        while(matcher.find()){
+            // Keep the whole placeholder for consistency with other formats, but also add just the placeholder name
+            // as this is allowed with the ICU format.
+            placeholder.add(matcher.group(1));
+            placeholder.add(matcher.group(1).replace("{", "").replace("}", "").trim());
```
[the remark this answers is not in the document]
Comment (Haehnchen) [reply src/main/java/fr/adrienbrault/idea/symfony2plugin/translation/dict/TranslationUtil.java]: i am not sure what is a better approach for now. so just put in both i suggest, maybe we get some feedback after we released it

**Comment by Haehnchen** [comment]
good work. i think navigation is not working, but should be in a new pr. i will merge this during this week.

navigation scope:
```
fr.adrienbrault.idea.symfony2plugin.translation.dict.TranslationUtil#getTranslationPsiElements
fr.adrienbrault.idea.symfony2plugin.translation.dict.TranslationUtil#getTargetForXlfAsXmlFile
```

**Comment by digilist** [comment]
Thx, I updated the code accordingly. I think, this can be merged then.

Regarding the navigation scope, I do not know yet what that is exactly, it's my first dive into IDEA Plugins. If you can give me some hints, I'd like to try to fix it :)

**Pull request merged**